### DISCUSSION TITLE
I have a modpack and I want to generate the files for this modpack's server

### DISCUSSION THREAD

**Discussion by IanSan19** [Griefed/ServerPackCreator#532] [Q&A] [answered]
Eu tenho um [modpack ](https://www.curseforge.com/minecraft/modpacks/dungeons-dragons-revolution) onde eu queria disponibilizar ele para servidores também, mas para isso eu preciso dos arquivos para servidor desse modpack, o suporte do CurseFoge me recomendou o seu aplicativo, mas eu não sei como usalo.
Você poderia me ajudar? Como faço para gerar o os arquivos para servidor desse meu modpack usando o seu aplicativo?

**Comment by Griefed** [comment, marked as answer]
Hi IanSan19,

in order to create a server pack from a modpack using ServerPackCreator, you need to:

1. head to https://github.com/griefed/serverpackcreator/releases/latest and download the installer for your OS. 
    1. Windows = "ServerPackCreator-VERSION-Installer-Windows-x86_64.exe"
    2. Linux = "ServerPackCreator-VERSION-Installer-Linux-amd64.sh"
    3. MacOS = "ServerPackCreator-VERSION-Installer-Mac.dmg"
2. Run the installer and start ServerPackCreator
3. The first time you start ServerPackCreator you are asked whether you want to go through a small step-by-step guide. I highly recommend you do that, as it will point you towards the most important parts of the configuration-process for your server pack.
4. Hit "Generate" once you are fine with your config :)

Some small hints:

- The modpack you select must be installed through a launcher like CurseForge, GDLauncher or ATLauncher. The ZIP-file downloaded from CurseForge does *not* work.
- After selecting your modpack-directory ServerPackCreator will try to determine the most important settings automatically. For example: Minecraft version, modloader and modloader version, as well as directories to include in your server pack.
- TEST YOUR SERVER PACK! Very important! Before you upload the generated server pack, run it on your machine and see whether the server runs as expected. Using ServerPackCreator is not a guarantee for a working server pack. You may still need to manually remove mods from the server, adjust configs, add files etc. etc.. 

Hope this helps!

Cheers
Griefed
